plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSystemTables.java

```diff
@@ -251,23 +266,162 @@ public void testManifestsTable()
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES (3, CAST('2019-09-09' AS DATE)), (4, CAST('2019-09-10' AS DATE)), (5, CAST('2019-09-10' AS DATE))", 3);
+
+            assertQuerySucceeds("SELECT * FROM \"" + table.getName() + "$files\" ");
+
+            assertQuerySucceeds("DELETE FROM " + table.getName() + " WHERE _bigint = 0");
+
+            MaterializedResult filePathResult = computeActual("SELECT file_path FROM \"" + table.getName() + "$files\" ORDER BY file_path");
+            long[] fileSizes = getFileSizes(filePathResult.getMaterializedRows());
+
+            MaterializedResult result = computeActual("SELECT content, file_format, file_size_in_bytes, record_count, column_sizes, value_counts, null_value_counts," +
+                    " nan_value_counts, key_metadata, split_offsets, equality_ids, lower_bounds, upper_bounds FROM \"" + table.getName() + "$files\" ORDER BY file_path");
+
+            MaterializedResult expectedStatistics =
+                    MaterializedResult.resultBuilder(
+                                    getSession(),
+                                    INTEGER,
+                                    VARCHAR,
+                                    BIGINT,
+                                    BIGINT,
+                                    new MapType(INTEGER, BIGINT, new TypeOperators()),
+                                    new MapType(INTEGER, BIGINT, new TypeOperators()),
+                                    new MapType(INTEGER, BIGINT, new TypeOperators()),
+                                    new MapType(INTEGER, BIGINT, new TypeOperators()),
+                                    VARBINARY,
+                                    new ArrayType(BIGINT),
+                                    new ArrayType(INTEGER),
+                                    new MapType(INTEGER, VARCHAR, new TypeOperators()),
+                                    new MapType(INTEGER, VARCHAR, new TypeOperators()))
+                            .row(0, "ORC", fileSizes[0], 2L, null, Map.of(1, Long.valueOf(2), 2, Long.valueOf(2)), Map.of(1, Long.valueOf(0), 2, Long.valueOf(0)), null, null, null, null, Map.of(1, "1", 2, "2019-09-09"), Map.of(1, "2", 2, "2019-09-09"))
+                            .row(0, "ORC", fileSizes[1], 1L, null, Map.of(1, Long.valueOf(1), 2, Long.valueOf(1)), Map.of(1, Long.valueOf(0), 2, Long.valueOf(0)), null, null, null, null, Map.of(1, "3", 2, "2019-09-09"), Map.of(1, "3", 2, "2019-09-09"))
+                            .row(0, "ORC", fileSizes[2], 2L, null, Map.of(1, Long.valueOf(2), 2, Long.valueOf(2)), Map.of(1, Long.valueOf(0), 2, Long.valueOf(0)), null, null, null, null, Map.of(1, "4", 2, "2019-09-10"), Map.of(1, "5", 2, "2019-09-10"))
+                            .build();
+
+            assertEquals(result, expectedStatistics);
+        }
+    }
```

Comment:
The above changes are unrelated to a new `$all_files` system table. Please extract a commit.